packages/block-editor/src/components/contrast-checker/index.js

```diff
@@ -50,11 +52,15 @@ function ContrastChecker( {
 			description: __( 'link color' ),
 		},
 	];
+	const strictContrastCheck = select( preferencesStore ).get(
```

Comment:
I just wanted to note that preferences for `core/edit-post` will be only available in the edit post screen, but never in other cases.

In general, in the block editor, we should rather pass down such setting so it could be read as any other block editor setting. @talldan and @youknowriad, how is it handled in other places?

Reply:
Apologies for the late response, have been AFK.

I think it's worth thinking about whether we want one WordPress-wide setting for this (i.e. if I toggle it on in the post editor, does it also become active in other editors?). There aren't any other user visible preferences that do this at the moment, they're all editor specific. But color contrast does seem like a policy-level thing for a user where they wouldn't really want this on in one editor an off in another, so perhaps this is the first preference where we should think about making this happen. What do you think?

If so, we can use a shared name like 'core' instead of 'core/edit-post'.

If not, then using an editor setting is the historical way to do this. The editor settings are often extended on initialisation of an editor, and some preferences become editor settings. Here's an example for the post editor:
https://github.com/WordPress/gutenberg/blob/a265d13b878cb2c7deaf91d3ee44212675fd2885/packages/edit-post/src/editor.js#L109-L156

The other editors are pretty much the same apart from the site editor, which does this in a selector. Doing it this way means the component can just read the editor setting without having to be concerned with which editor is active.

Reply:
The other issue you mentioned about needing to refresh might be because this is `select` and not `useSelect`.

Reply:
Thank you for the advice on using `useSelect`, that worked 👍 

> color contrast does seem like a policy-level thing for a user where they wouldn't really want this on in one editor an off in another

It seems to already work fine on my end... 🤔 
To test this, I added in https://github.com/WordPress/gutenberg/pull/41363/files#diff-6bf1ecd7162a5f625e50d3c434dec2da49ee8143f738358c4c02f4c537af73afR75 `console.log( isReadableOptions.level )`.
In the post-editor, I changed the preference to AAA. Then checked the post-editor and the site-editor and both were using `AAA` as expected.
I then switched back to AA, and sure enough, both the site-editor and post-editor were using AA for the contrast-checks.
Am I missing something?

Reply:
Hmmmm should we use `'core'`, or would something like `'core/global'` be more suitable? 🤔

Reply:
I think just 'core' is fine.